### PR TITLE
Replace usage of OSCoreCtx(from cf) by OscoreSeting in SecurityInfo

### PR DESCRIPTION
**Pull request by sbernard31**
I cherry-pick some @Michal-Wadowski's improvement from #1175.
Which should allow to start to work on support of Oscore in server demos. (#1204)

### CODE REVIEW

File: leshan-server-core/src/main/java/org/eclipse/leshan/server/security/oscore/OscoreSetting.java [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+        return masterSecret;
+    }
+
+    public Integer getAeadAlgorithm() {
+        return aeadAlgorithm;
+    }
+
+    public Integer getHmacAlgorithm() {
+        return hmacAlgorithm;
+    }
```
Comment (sbernard31) [inline leshan-server-core/src/main/java/org/eclipse/leshan/server/security/oscore/OscoreSetting.java:61]: Maybe better to replace `Integer` by dedicated class/enum, like I did in : https://github.com/eclipse/leshan/pull/1204/files#diff-593f0c45e56ce01047c78d607542064f008dda0112d69f3c3097a237c5a0e32cR26

File: leshan-server-core/src/main/java/org/eclipse/leshan/server/security/SecurityInfo.java [hunk 2]
```diff
@@ -114,26 +121,42 @@ public static SecurityInfo newX509CertInfo(String endpoint) {
...
-        db.addContext(oscoreCtx);
+        db.addContext(getContext(oscoreSetting));
 
-        return new SecurityInfo(endpoint, null, null, null, false, oscoreCtx);
+        return new SecurityInfo(endpoint, null, null, null, false, oscoreSetting);
+    }
+
+    private static OSCoreCtx getContext(OscoreSetting oscoreSetting) {
+        try {
+            OSCoreCtx osCoreCtx = new OSCoreCtx(oscoreSetting.getMasterSecret(), true,
```
Comment (sbernard31) [inline leshan-server-core/src/main/java/org/eclipse/leshan/server/security/SecurityInfo.java:138]: I'm not sure to understand the second argument (boolean) ?
The javadoc says: 
```
 * @param client is this originally the client's context
```
But still not clear to me, I don't know if I should use true or false. (with true or false integration tests seems to be happy :thinking:  )

Comment (Michal-Wadowski) [reply leshan-server-core/src/main/java/org/eclipse/leshan/server/security/SecurityInfo.java:138]: According to the constructor of OSCoreCtx the ```boolean client``` only matters in the case if sender_id or recipient_id is not defined.

Comment (rikard-sics) [reply leshan-server-core/src/main/java/org/eclipse/leshan/server/security/SecurityInfo.java:138]: Yes, indeed. That parameter is only relevant if the sender ID and recipient ID is not defined. I think the original intention was for an application the be able to create a very simple OSCORE context for testing.

**Comment by sbernard31** [comment]
This is now integrated in `oscore` branch (commit 992b356)

**Pull request closed without merging**